assets/components/ReportCardContainer.js

```diff
@@ -33,17 +34,196 @@ class ReportCardContainer {
+    let galleryContainer = document.createElement('div');
+    galleryContainer.setAttribute('class', 'report-card__image-gallery');
     let gallery = document.createElement('div');
-    gallery.setAttribute('id', 'report-card__picture-container');
+    gallery.setAttribute('class', 'report-card__picture-container');
 
     for (const pic of photos) {
       let image = document.createElement('img');
       image.src = pic;
       gallery.appendChild(image);
     }
-    return gallery;
+    galleryContainer.appendChild(gallery);
+    return galleryContainer;
+  }
+
+  getDescription() {
+    const contentHTML = `	
+    <p class="text-body-3 regular">Description</p>
+		<p class="text-body-2 regular">${this.comment}</p>`;
+    const div = document.createElement('div');
+    div.setAttribute('class', 'report-card__description');
+    div.innerHTML = contentHTML;
+    return div;
+  }
+  getMyReportButtons() {
+    const contentHTML = `	
+    ${ToggleSwitch(this.id, !this.deleted_at)}
+    <button class="btn btn-tertiary text-body-3 medium" onclick="window.location.href='/pages/report-hazard/index.html?id=${this.id}#review-report'">
+      <i class="icon-edit"></i>
+      Edit Report
+    </button>`;
+    const div = document.createElement('div');
+    div.setAttribute('class', 'report-card__my-reports-buttons');
+    div.innerHTML = contentHTML;
+    return div;
+  }
+
+  getReportedBy() {
+    const contentHTML = `
+    <p class="text-body-3 regular">
+    Reported by ${this.createdByUserLoggedIn ? 'You' : ''}
+  </p>
+  
+  ${
+    !this.createdByUserLoggedIn
+      ? `
+  <div class="report-card__user-details">
+    <img
+      id="user-image"
+      src="${this.user.photo || '../../assets/img/default-nav-image.png'}"
+      alt="User photo"
+    />
+    <p class="text-body-2 regular">${this.user.name || 'Anonymous user'}</p>
+  </div>
+  `
+      : ''
+  }
+  
+  <div
+    class="${this.createdByUserLoggedIn ? 'hidden' : ''}"
+  ></div>`;
+    const div = document.createElement('div');
+    div.setAttribute('class', 'report-card__reported-by');
+    div.innerHTML = contentHTML;
+    return div;
+  }
+
+  getReportFlagButtons() {
+    const contentHTML = `
+     <button class="btn btn-success" id="stillThereBtn">
+      <i class="icon-check"></i>
+      Still there
+    </button>
+
+    <button class="btn btn-warning" id="notThereBtn">
+      <i class="icon-close"></i>
+      Not there
+    </button>
+
+    <button class="btn btn-error" id="flagReportBtn">
+      <i class="icon-flag"></i>
+      Flag report
+    </button>
+  
+    <div id="flagReportMessage" class="message error">
+      <i class="icon-flag message__icon"></i>
+      <p class="message__content text-body-3 medium">
+        You have flagged this report as fake.
+      </p>
+    </div>
+    `;
+    const div = document.createElement('div');
+    div.setAttribute('class', "report-card__hazard-detail-buttons ${this.createdByUserLoggedIn ? 'hidden' : ''}");
```

Comment:
Does this need to be inside backticks?
```
"report-card__hazard-detail-buttons ${this.createdByUserLoggedIn ? 'hidden' : ''}"
```

Reply:
thanks! https://github.com/whamester/WMDD-4885-Project/pull/122/commits/6c47bbcbff6ca76a64e9ecb9aa9e812b90ca0a40